feat: add internal/external Postgres config

pmoscode opened the pull request:
## Description

Adds the ability to configure the PostgreSQL DB. Now, you can bring your own PostgreSQL or use the bundled one.
And you can also predefine the DB password or let PostgreSQL create one for you.

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up to date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] Copyright and license header are present on all affected files

pmoscode commented:
@borisrizov-zf @mknoopvw 
Please, review the changes.

pmoscode commented:
@SebastianBezold 
Please merge this PR after review is done.
Thx

borisrizov-zf commented on charts/managed-identity-wallet/tests/default/deployment_test.yaml:
```diff
@@ -99,7 +99,7 @@ tests:
       - lengthEqual:
           path: spec.template.spec.containers[0].env
-          count: 1
+          count: 9
```
We shouldn't use magic numbers in tests, add a comment describing the origin of `9`

pmoscode replied:
Done. And fixed a small bug. :)

borisrizov-zf commented on charts/managed-identity-wallet/values.yaml:
```diff
@@ -23,40 +23,53 @@ replicaCount: 1
-#  DB_PASSWORD: xxx # Has to be the same as below in "postgresql" section
-#  KEYCLOAK_CLIENT_ID: xxx
-#  ENCRYPTION_KEY: xxx  # 64 chars
+# -- Parameters for the application (will be stored as secrets - so, for important values, ...)
```
The sentence is confusing. Please reword.

pmoscode replied:
Done.

SebastianBezold commented:
HI all,
there are again valuable comments to the change from @borisrizov-zf, but the PR is approved anyways. 
Just as suggestion, I would rather address or fix the comments right in this PR (preferred), or at least create issues as a reminder of the spots, that can be improved.
I think if i'll merge the PR without addressing the comments or creating issues, the valuable feedback will be lost

pmoscode commented on .github/workflows/chart-lint.yml:
```diff
@@ -1,68 +1,87 @@
+     # Setup python as a prerequisite for chart linting
+     - uses: actions/setup-python@v2
+       with:
+         python-version: 3.7
```
Done.

pmoscode commented on charts/managed-identity-wallet/values.yaml:
```diff
@@ -23,40 +23,53 @@ replicaCount: 1
+secrets:
+  # -- Keycloak confidential client id for MIW application
+  KEYCLOAK_CLIENT_ID: ""
+  # -- AES encryption key used to encrypt/decrypt private keys (random 32 chars)
```
Done.

mknoopvw commented:
I am just curious:
In the "{stage}-values.yaml" were "" added for some values: is this to fix the fault or just to unify the appearance?
Thank you.

borisrizov-zf commented on charts/managed-identity-wallet/values-dev.yaml:
```diff
@@ -21,16 +21,13 @@ image:
-  DB_USER_NAME: <path:managed-identity-wallets/data/dev/postgres-managed-identity-wallets-secret-config#user>
-  DB_PASSWORD: <path:managed-identity-wallets/data/dev/postgres-managed-identity-wallets-secret-config#password>
-  ENCRYPTION_KEY: 76a7834fb37e090c2789a9b1a76748d3
+  ENCRYPTION_KEY: "76a7834fb37e090c2789a9b1a76748d3"
```
According to yaml, no. But it's a good idea when special characters might appear in the value.

pmoscode commented:
> HI all,
> there are again valuable comments to the change from @borisrizov-zf, but the PR is approved anyways.
> Just as suggestion, I would rather address or fix the comments right in this PR (preferred), or at least create issues as a reminder of the spots, that can be improved.
> I think if i'll merge the PR without addressing the comments or creating issues, the valuable feedback will be lost

Hi @SebastianBezold,
sorry for the misleading post. First all reviews should be green, and then...
Now, it's done. 
Fyi: The "Lint and Test Chart" check is failing, because the image isn't available on hub.docker.com yet.

Thx

pmoscode commented:
> > > HI all,
> > > there are again valuable comments to the change from @borisrizov-zf, but the PR is approved anyways.
> > > Just as suggestion, I would rather address or fix the comments right in this PR (preferred), or at least create issues as a reminder of the spots, that can be improved.
> > > I think if i'll merge the PR without addressing the comments or creating issues, the valuable feedback will be lost
> > 
> > 
> > Hi @SebastianBezold, sorry for the misleading post. First all reviews should be green, and then... Now, it's done. Fyi: The "Lint and Test Chart" check is failing, because the image isn't available on hub.docker.com yet.
> > Thx
> 
> Hi @pmoscode! The "Lint and Test Charts" workflow fails because one of the container environment variables in the `values.yaml` is invalid. See the [error description](https://github.com/eclipse-tractusx/managed-identity-wallet/actions/runs/5794708643/job/15704829066?pr=48#step:13:33).

Yes, you're right. Fixed it. But the test is still failing, because the corresponding image isn't released on Docker Hub yet. Pipeline is still in dev.

pmoscode commented:
@almadigabor 
Fix is done.

almadigabor commented:
Merged this PR but raised a new issue #51, so the helm test workflow can run properly even when the new image version is not yet released.
